core/src/main/java/org/microshed/testing/jaxrs/RestClientBuilder.java

```diff
@@ -176,33 +169,33 @@ private static String locateApplicationPath(Class<?> clazz) {
                 String checkPkg = pkgs[0] + '.' + pkgs[1] + '.' + pkgs[2];
                 LOG.debug("checking in pkg: " + checkPkg);
                 appClasses = ReflectionSupport.findAllClassesInPackage(checkPkg,
-                                                                       c -> Application.class.isAssignableFrom(c) &&
-                                                                            AnnotationSupport.isAnnotated(c, ApplicationPath.class),
-                                                                       n -> true);
+                        c -> Application.class.isAssignableFrom(c) &&
+                                AnnotationSupport.isAnnotated(c, ApplicationPath.class),
+                        n -> true);
             }
         }
 
-        if (appClasses.size() == 0) {
+        if (appClasses.isEmpty()) {
             LOG.info("No classes implementing 'jakarta.ws.rs.core.Application' found on classpath to set base path from " + clazz +
-                     ". Defaulting base path to '/'");
+                    ". Defaulting base path to '/'");
             return "";
         }
 
         Class<?> selectedClass = appClasses.stream()
-                        .sorted((c1, c2) -> c1.getName().compareTo(c2.getName()))
-                        .findFirst()
-                        .get();
+                .sorted(Comparator.comparing(Class::getName))
+                .findFirst()
+                .get();
         ApplicationPath appPath = AnnotationSupport.findAnnotation(selectedClass, ApplicationPath.class).get();
         if (appClasses.size() > 1) {
             LOG.warn("Found multiple classes implementing 'jakarta.ws.rs.core.Application' on classpath: " + appClasses +
-                     ". Setting base path from the first class discovered (" + selectedClass.getCanonicalName() + ") with path: " +
-                     appPath.value());
+                    ". Setting base path from the first class discovered (" + selectedClass.getCanonicalName() + ") with path: " +
+                    appPath.value());
         }
         LOG.debug("Using base ApplicationPath of '" + appPath.value() + "'");
         return appPath.value();
     }
 
-    private static String join(String firstPart, String secondPart) {
+    private static String joinAppAndJaxrsPath(String firstPart, String secondPart) {
```

Comment:
If we are making this method more clear perhaps switching the parameters would be better.
```suggestion
    private static String join(String appContextRoot, String jaxrsPath) {
```

Reply:
Wouldn't agree as the name join clashes with java's own join function. Changing the name makes this distinctive.

Reply:
```suggestion
    private static String joinPaths(String appContextRoot, String jaxrsPath) {
```
Perhaps this would be a good compromise.

Reply:
Done!